src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -281,17 +284,68 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
     /// <summary>
     /// Return the first verified dapr path
     /// </summary>
-    static string? GetDefaultDaprPath()
+    async Task<string?> GetDefaultDaprPathAsync(CancellationToken cancellationToken)
     {
-        foreach (var path in GetAvailablePaths())
+        if (_defaultDaprPath is null)
         {
-            if (File.Exists(path))
+            await _daprPathLock.WaitAsync(cancellationToken).ConfigureAwait(false);
+
+            try
             {
-                return path;
+                if (_defaultDaprPath is null)
+                {
+                    Process? process = null;
+
+                    try
+                    {
+                        process = Process.Start(
```

Comment:
Can you do this instead?

https://github.com/dotnet/aspire/blob/31ce95ba73ec2b526e90a7ddef795bb79627c95f/src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs#L398-L420